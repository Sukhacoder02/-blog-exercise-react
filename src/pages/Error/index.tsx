import * as React from 'react';
import { Navbar } from '../../components';
import { Footer } from '../../components';
import { useParams } from 'react-router-dom';
import './Error.css';

const Error: React.FC = (): JSX.Element => {
  const { statusCode } = useParams();
  return (
    <div className="body">
      <Navbar />
      <div id="main">
        <div className="errorDiv">
          <h2>Something went wrong</h2>
          {statusCode && <p>HTTP Status Code: {statusCode}</p>}
        </div>
      </div>
      <Footer />
    </div>
  );
};

export default Error;
