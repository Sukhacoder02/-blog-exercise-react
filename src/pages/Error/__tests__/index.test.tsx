import * as React from 'react';
import { render, screen } from '@testing-library/react';
import Error from '..';
import { useParams } from 'react-router-dom';

jest.mock('react-router-dom');
describe('Page Error', () => {
  const mockedUseParams = useParams as jest.MockedFunction<typeof useParams>;
  afterEach(() => {
    jest.clearAllMocks();
  });
  it('Should render without crashing', () => {
    mockedUseParams.mockReturnValueOnce({ statusCode: '404' });
    const { asFragment } = render(<Error />);
    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByText('Something went wrong')).toBeTruthy();
  });
  it('Should display the status code when it is passed in the parans', () => {
    mockedUseParams.mockReturnValueOnce({ statusCode: '404' });
    render(<Error />);
    expect(screen.getByText('HTTP Status Code: 404')).toBeTruthy();
  });
});
