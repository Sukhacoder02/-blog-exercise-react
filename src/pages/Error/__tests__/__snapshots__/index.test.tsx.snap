// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Error Should render without crashing 1`] = `
<DocumentFragment>
  <div
    class="body"
  >
    <div>
      <header>
        <div
          class="heading"
        >
          <h2>
            The Artifact
          </h2>
          <h4>
            <em>
               Culture 
            </em>
            & 
            <em>
              Art
            </em>
             Blog
          </h4>
        </div>
      </header>
      <div
        class="navBar"
      >
        <nav>
          <a
            href="#"
          >
            Blog
          </a>
          <a
            href="#"
          >
            About
          </a>
          <a
            class="contactLink"
            href="#"
          >
            Contact
          </a>
        </nav>
      </div>
    </div>
    <div
      id="main"
    >
      <div
        class="errorDiv"
      >
        <h2>
          Something went wrong
        </h2>
        <p>
          HTTP Status Code: 404
        </p>
      </div>
    </div>
    <footer>
      <div
        class="footerHeading"
      >
        <h4>
          @artifact.com 2023
        </h4>
      </div>
    </footer>
  </div>
</DocumentFragment>
`;
